src/tool/xmeta/cpp/models/property_model.h

```diff
@@ -57,9 +69,38 @@ namespace xlang::xmeta
             return m_set_method;
         }
 
+        auto const& get_semantic() const noexcept
+        {
+            return m_semantic;
+        }
+
+        void create_get_method()
+        {
+            m_get_method = std::make_shared<method_model>(
+                "get_" + get_id(),
+                get_decl_line(),
+                get_assembly_name(),
+                type_ref{ m_type },
+                std::vector<formal_parameter_model>{},
+                method_semantics{});
+        }
+
+        void create_set_method()
+        {
+            m_set_method = std::make_shared<method_model>(
+                "set_" + get_id(),
+                get_decl_line(),
+                get_assembly_name(),
+                type_ref{ m_type },
+                std::vector<formal_parameter_model>{},
+                method_semantics{});
+        }
+
     private:
+        type_ref m_type;
+        bool m_is_array;
         property_semantics m_semantic;
-        property_type_semantics m_type;
+        bool m_get_declared_first;
```

Comment:
This member is of dubious value. What we *really* care about is the ordering of the methods.